core/src/main/java/org/elasticsearch/indices/memory/IndexingMemoryController.java

```diff
@@ -154,18 +161,21 @@ public IndexingMemoryController(Settings settings, ThreadPool threadPool, Indice
         // we need to have this relatively small to move a shard from inactive to active fast (enough)
```

Comment:
Pre-existing issue: I think 30s default is too large, because an index that was inactive and suddenly becomes active with a 512 KB indexing buffer for up to 30s of heavy indexing is suddenly writing many, many segments.  It would be better if the first indexing op to arrive to an inactive shard could force IMC to wake up and re-evaluate.

 I'll open a separate issue about this ...